framework/wazuh/core/utils.py

```diff
@@ -940,6 +940,32 @@ def xml_to_dict(conf, section_name):
             raise WazuhError(1127, extra_message=f"global > limits > {disabled_limit}")
 
 
+def check_agents_versions(data: str):
+    """Check if higher version agents are allowed.
+    If not, it will check if the agent version is in the list of exceptions.
+
+    Parameters
+    ----------
+    data : str
+        Configuration file content
+    """
+    blocked_configurations = configuration.api_conf['upload_configuration']
+
+    def check_section(versions_regex, split_section):
+        try:
+            for line in versions_regex.findall(data)[0].split(split_section):
+                versions_matches = re.match(r".*<version>(.*)</version>.*", line, flags=re.MULTILINE | re.DOTALL)
```

Comment:
I didn't see any reference to this `<version>` field either in the [original issue](https://github.com/wazuh/wazuh/issues/18138) or [in the API](https://github.com/wazuh/wazuh/issues/18139) or [QA tests issue](https://github.com/wazuh/wazuh-qa/issues/4371). Did you consult the core team or how do you know of its existence?

The purpose of the issue is, if enabled, to prevent a user from uploading configurations containing the `<allow_higher_versions>yes</allow_higher_versions>` block via the API. This does not have access to the agents' version (each agent could have a different version), only to the manager's `ossec.conf`.

Therefore, I think it doesn't make sense to look for exceptions: either the manager allows agents with a higher version to connect or does not allow it. And for the API, the decision isn't even that, but whether to upload an `ossec.conf` containing `<allow_higher_versions>yes</allow_higher_versions>` is allowed or not.

Reply:
I included the `<version>` tag because the issue linked to this pull request mentioned an `exceptions` array and we needed a way to specify those in the configuration (I thought we would allow users to reject higher versions except the ones they specified in the exceptions). Since the Core team did not start the development yet, I thought I would just use `version` initially and discuss that with them later.

In any case, if there are no exceptions needed, I'll remove them right away and just keep `allow_higher_versions`